flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExceptionHistoryEntry.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.runtime.executiongraph.AccessExecution;
+import org.apache.flink.runtime.executiongraph.ErrorInfo;
+import org.apache.flink.runtime.taskmanager.TaskManagerLocation;
+
+import javax.annotation.Nullable;
+
+/**
+ * {@code ExceptionHistoryEntry} collects information about a single task failure that should be
+ * exposed through the exception history.
+ */
+public class ExceptionHistoryEntry extends ErrorInfo {
+
+    private static final long serialVersionUID = -3855285510064263701L;
+
+    @Nullable private final String failingTaskName;
+    @Nullable private final TaskManagerLocation assignedResourceLocation;
```

Comment:
Interesting finding. I'm wondering whether it would make sense to create a `TaskManagerLocation` interface since we have the same issue with the `ArchivedExecutionVertex`.

Reply:
I created a `ArchivedTaskManagerLocation` for now as the refactoring would take a larger effort which is out of scope for this ticket. @zentol Do we already have a ticket that addresses this refactoring? If not, whats the headline of the refactoring ticket I would create? I understood that introducing `TaskManagerLocation` interface was part of a larger effort?